ReactiveSwift.playground/Pages/Sandbox.xcplaygroundpage/Contents.swift

```diff
@@ -22,4 +22,186 @@ import Foundation
  A place where you can build your sand castles 🏖.
 */
 
+enum PromiseResult<Value, Error: Swift.Error> {
```

Comment:
We could just use `Result` for this.

Reply:
Well that was my intention in the first place, but then I couldn't work around the case with `nil` error.
That's actually a questionable decision to store it this way, other implementations chose to wrap a user error into additional error enum like

```
enum PromiseError: Error {
     case nested(error)
     case interrupted
}
```

In this case standard `Result` would do. I'm open for suggestions here - including an option to `fatalError` the case of interrupted / empty source producer. Which is of course not something I'd agree for ;)